frontend/src/components/HistorySection/HistorySection.tsx

```diff
@@ -47,52 +47,79 @@ const HistorySection = () => {
   }
 
   return (
-    <div className="h-full flex flex-col p-4 gap-4 bg-background border-t-2">
-      <h1 className="text-xl font-semibold">History</h1>
-      {history.length > 0 ? (
-        <Carousel
-          opts={{
-            align: "start"
-          }}
-          className="w-[80%] self-center"
-        >
-          <CarouselContent>
-            {history.map(history => (
-              <CarouselItem
+    <TooltipProvider>
+      <div className="h-full flex flex-col p-4 gap-4 bg-background border-t-2">
+        <h1 className="text-xl font-semibold">History</h1>
+        {history.length > 0 ? (
+          <div className="flex gap-4 flex-row w-full overflow-auto">
+            {history.map((history, idx) => (
+              <HistoryItem
                 key={history.id}
-                className="md:basis-1/2 lg:basis-1/3"
+                history={history}
+                idx={idx}
                 onClick={() => handleHistoryItemClick(history)}
-              >
-                <Card className="max-w-[6rem]">
-                  <CardContent className="flex aspect-square items-center justify-center p-2 text-center w-full h-full relative bg-muted">
-                    {history.imageResult ? (
-                      <Image
-                        fill
-                        objectFit="cover"
-                        src={history.imageResult}
-                        alt="test"
-                        className="p-2 rounded-md object-cover cursor-pointer"
-                      />
-                    ) : (
-                      <Image
-                        fill
-                        objectFit="contain"
-                        src="/asset/noImgErr.webp"
-                        alt="No Image found"
-                      />
-                    )}
-                  </CardContent>
-                </Card>
-              </CarouselItem>
+              />
             ))}
-          </CarouselContent>
-          <CarouselPrevious />
-          <CarouselNext />
-        </Carousel>
-      ) : (
-        <p>No history recorded</p>
-      )}
-    </div>
+          </div>
+        ) : (
+          <p>No history recorded</p>
+        )}
+      </div>
+    </TooltipProvider>
+  )
+}
+
+const HistoryItem = ({
+  history,
+  onClick,
+  idx
+}: {
+  history: History
+  onClick: () => void
+  idx: number
+}) => {
```

Comment:
This is ```HistorySection``` child, should it be another file as our pattern?
Assign this component with type, it would be more readable and maintainable when this type growth. The exception is when component's props quantity is one.
Update index.ts
```suggestion
const HistoryItemType = {
  history: History
  onClick: () => void
  idx: number
}

const HistoryItem:FC<HistoryItemType> = ({
  history,
  onClick,
  idx
}) => {
```